src/vg_protocol.erl

```diff
@@ -125,69 +125,88 @@ encode_bytes(undefined) ->
 encode_bytes(Data) ->
     [<<(size(Data)):32/signed-integer>>, Data].
 
-encode_record_set([]) ->
-    [];
-encode_record_set(Record) when is_binary(Record) ->
-    Record2 = encode_record(Record),
-    [<<0:64/signed-integer, (iolist_size(Record2)):32/signed-integer>>, Record2];
-encode_record_set([Record | T]) ->
-    Record2 = encode_record(Record),
-    [<<0:64/signed-integer, (iolist_size(Record2)):32/signed-integer>>, Record2 | encode_record_set(T)].
-
-encode_replicate_record_set([]) ->
+record_timestamp_or_now(#{timestamp := Timestamp}) when is_integer(Timestamp) ->
+    Timestamp;
+record_timestamp_or_now(_) ->
+    erlang:system_time(millisecond).
+
+sequence_or_zero(#{sequence_number := SequenceNumber}) when is_integer(SequenceNumber) ->
+    SequenceNumber;
+sequence_or_zero(_) ->
+    0.
+
+encode_record_batch(Records) ->
+    encode_record_batch(0, 0, Records).
+
+encode_record_batch(_, _, []) ->
```

Comment:
this is not exported, is there any way for anything to specify producer id and epoch right now?